src/server/api/routers/media-request.ts

```diff
@@ -147,6 +149,16 @@ const constructAvatarUrl = (appUrl: string, avatar: string) => {
   return `${appUrl}/${avatar}`;
 };
 
+const constructfallbackAvatarUrl = (appUrl: string, avatar: string) => {
+  const isAbsolute = avatar.startsWith('http://') || avatar.startsWith('https://');
+
+  if (isAbsolute) {
+    return avatar;
+  }
+
+  return `${appUrl}/avatarproxy/${avatar}`;
+};
+
```

Comment:
This is just the same function as constructAvatarUrl with just one small exception. You should adapt the original one instead.

```suggestion
const constructAvatarUrl = (appUrl: string, avatar: string, path?: string) => {
  const isAbsolute = avatar.startsWith('http://') || avatar.startsWith('https://');

  if (isAbsolute) {
    return avatar;
  }

  return `${appUrl}/${path?.concat("/") ?? ""}${avatar}`;
};

```

(I'm just writing this on the fly, double check it first)

Reply:
I tested it and it works fine